src/core/accumulators/Correlator.cpp

```diff
@@ -91,13 +91,13 @@ std::vector<double> componentwise_product(std::vector<double> const &A,
 
 std::vector<double> tensor_product(std::vector<double> const &A,
                                    std::vector<double> const &B,
-                                   Utils::Vector3d) {
+                                   Utils::Vector3d const &) {
   std::vector<double> C(A.size() * B.size());
   auto C_it = C.begin();
 
-  for (auto A_it = A.begin(); A_it != A.begin(); ++A_it) {
-    for (double B_it : B) {
-      *(C_it++) = *A_it * B_it;
+  for (double a : A) {
```

Comment:
Is there a specific reason why you can't just say `boost::transform(A, B, C.begin(), std::multiplies<>{})`?
(In this case a would even consider using an `std::back_inserter` instead of explicitly resizing the vector,
and doing `boost::transform(A, B, std::back_inserter(C), std::multiplies<>{})`)

Reply:
Ah sorry, it really is the tensor product.  It's too late.